ambry-api/src/main/java/com/github/ambry/quota/QuotaSource.java

```diff
@@ -22,16 +23,71 @@
  * usage of a resource can be saved.
  */
 public interface QuotaSource {
+
+  /**
+   * Method to initialize the {@link QuotaSource}.
+   * @throws QuotaException in case of any exception.
+   */
+  void init() throws QuotaException;
+
+  /**
+   * @return {@code true} if the {@link QuotaSource} is initialized and ready to be used. {@code false} otherwise.
+   */
+  boolean isReady();
+
   /**
    * Get the {@link Quota} for specified resource and operation.
    * @param quotaResource {@link QuotaResource} object.
    * @param quotaName {@link QuotaName} object.
+   * @throws QuotaException in case of any exception.
+   */
+  Quota getQuota(QuotaResource quotaResource, QuotaName quotaName) throws QuotaException;
+
+  /**
+   * Get the percent quota usage for the specified resource and operation.
+   * @param quotaResource {@link QuotaResource} object.
+   * @param quotaName {@link QuotaName} object.
+   * @return usage of quota in percentage.
+   * @throws QuotaException in case of any exception.
+   */
+  float getUsage(QuotaResource quotaResource, QuotaName quotaName) throws QuotaException;
```

Comment:
is there a reason that chargeUsage takes in doubles and getUsage returns float as opposed to using the same precision everywhere?

Reply:
So charge taken in double because the interface doesn't want to make assumptions about how large a cost could be. `getUsage` returns float, because it returns percentage usage.